src/compat-shallow-renderer/Preact10ShallowDiff.ts

```diff
@@ -0,0 +1,378 @@
+function installOptionsForShallowDiff() {
+  if (shallowOptionsInstalled) {
+    return;
+  }
+
+  /**
+   * Eagerly set _preact10ShallowDiff on the component instance so later code
+   * (e.g. the diffed option later in this function) can inspect it to know that
+   * this component is being shallow rendered and can respond accordingly
+   */
+  const prevRender = (options as any).__r;
+  (options as any).__r = (vnode: VNode<any>) => {
+    prevRender(vnode);
+
+    const component = getComponent(vnode) as ShallowDiffComponent | null;
+    if (component && Preact10ShallowDiff.current) {
+      component._preact10ShallowDiff = Preact10ShallowDiff.current;
+    }
+  };
+
+  /**
+   * If this component is being shallow rendered, remove all hook effect
+   * callbacks to match how enzyme shallow render's React
+   */
+  const prevDiffed = options.diffed;
+  options.diffed = vnode => {
+    const component = getComponent(vnode) as ShallowDiffComponent | null;
+    if (component && component._preact10ShallowDiff) {
+      removeEffectCallbacks(vnode as any);
+    }
+
+    prevDiffed?.(vnode);
+  };
+
+  shallowOptionsInstalled = true;
+}
+
+/**
+ * Replace existing setState and forceUpdate implementations with new
+ * implementations that detect if a component has been shallow rendered and
+ * ensures renders caused by setState/forceUpdate are also shallow rendered.
+ *
+ * Here, the _preact10ShallowDiff property does something similar as React
+ * Component's `updater` property. React Component's `updater` property contains
+ * a reference to the implementations of `setState`, `forceUpdate`, etc. and it
+ * is set by the code that rendered the component (e.g. ReactDOM). Here the
+ * presence of the `_preact10ShallowDiff` property (set by the shallow renderer)
+ * indicates `setState` and `forceUpdate` should call into our custom shallow
+ * implementation.
+ */
+function installShallowComponentHooks() {
+  const prevSetState = Component.prototype.setState;
+  Component.prototype.setState = function (
+    this: ShallowDiffComponent,
+    update: any,
+    callback: () => void
+  ) {
+    if (this._preact10ShallowDiff) {
+      shallowSetState.call(this as any, update, callback);
+    } else {
+      prevSetState.call(this, update, callback);
+    }
+  };
+
+  const prevForceUpdate = Component.prototype.forceUpdate;
+  Component.prototype.forceUpdate = function (
+    this: ShallowDiffComponent,
+    callback: () => void
+  ) {
+    if (this._preact10ShallowDiff) {
+      shallowForceUpdate.call(this as any, callback);
+    } else {
+      prevForceUpdate.call(this, callback);
+    }
+  };
+}
+
+installShallowComponentHooks();
+
+/**
+ * This class mirrors ReactShallowRenderer for the purpose of shallow rendering
+ * Preact components. It implements the same API and passes almost the exact
+ * same test suite
+ *
+ * https://github.com/enzymejs/react-shallow-renderer
+ */
+export default class Preact10ShallowDiff {
```

Comment:
Yea, I was thinking something similar, but originally went with this name to help differentiate it from the enzyme "renderers" (i.e. `MountRenderer`, `ShallowRenderer`, etc. that implement the enzyme renderer contract).

But I was thinking that before we put this file in its own folder with a readme, etc. So that helps clarify the difference between the classes. I'll do what you suggest here and rename this to `PreactShallowRenderer`

Reply:
I think I got everything. Renamed classes, properties, files, and comments to use the new name. I didn't see any other related occurrences using "find all", but let me know if you spot something I missed.